### PR TITLE
Add Javascript example to connecto to RDS using Lambda with IAM auth

### PR DESCRIPTION
**Pull request by Sizk**
*Issue #, if available:*

*Description of changes:*
Added JavaScript code snippet to connect to RDS using Lambda with IAM auth.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lambda-function-connect-rds-iam/example.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+async function dbOps() {
+
+  try {
+      // Obtain auth token
+      const token = await createAuthToken();
+      // Define connection configuration
+      let connectionConfig = {
+        host: process.env.ProxyHostName,
+        user: process.env.DBUserName,
+        //password: token,
```
Comment (kathvish) [inline lambda-function-connect-rds-iam/example.js:40]: What happened to passing the token as password? It should have been the most straightforward way to pass the token. If it didn't work, it's worth investigating why and leaving a comment about it if necessary.

File: lambda-function-connect-rds-iam/example.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        database: process.env.DBName,
+        ssl: 'Amazon RDS',
+        authPlugins: {
+          mysql_clear_password: () => () => Buffer.from(token)
```
Comment (kathvish) [inline lambda-function-connect-rds-iam/example.js:44]: Why do we need this? Would be good to have a comment explaining why.

**Review by kathvish — CHANGES_REQUESTED** [review]
Nice work! But please take a look at my comments and update accordingly, especially about error handling and passing the token to the DB connection.

File: lambda-function-connect-rds-iam/snippet-data.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    }
+  },
+  "snippets": [
+    {
+      "title": "Runtimes",
+      "codeTabs": [
+        {
+          "id": "Javascript",
+          "title": "Copy the code into your Lambda function",
+          "description": "Follow the AWS Lambda instructions for packaging with dependencies and uploading in .zip format. This snippet uses the ioredis library.",
```
[the remark this answers is not in the document]
Comment (Sizk) [reply lambda-function-connect-rds-iam/snippet-data.json]: Modified

File: lambda-function-connect-rds-iam/example.js (hunk 2 above)
Comment (Sizk) [reply lambda-function-connect-rds-iam/example.js]: Removed in favor of token

File: lambda-function-connect-rds-iam/example.js (hunk 1 above)
Comment (Sizk) [reply lambda-function-connect-rds-iam/example.js]: Removed in favor of token

File: lambda-function-connect-rds-iam/example.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      }
+      // Create the connection to the DB
+      const conn = await mysql.createConnection(connectionConfig);
+      // Obtain the result of the query
+      const [res,] = await conn.execute('select ?+? as sum', [2, 2]);
+      return res
+      
+  } catch (error) {
+      console.error(error)
+      return error
```
[the remark this answers is not in the document]
Comment (Sizk) [reply lambda-function-connect-rds-iam/example.js]: Removed catching altogether

File: lambda-function-connect-rds-iam/snippet-data.json [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  "title": "Connect to Amazon RDS from AWS Lambda",
+  "description": "This snippet demonstrates how to connect to an  Amazon Relational Database cluster from an AWS Lambda function with IAM authentication. This snippet is valid for version/engines XXXXXXXXXX",
+  "type": "Integration",
+  "services": ["lambda", "rds"],
+  "tags": [],
+  "languages": ["Javascript"],
+  "introBox": {
+    "headline": "How it works",
+    "text": [
+      "Amazon Relational Database Service (Amazon RDS) is a collection of managed services that makes it simple to set up, operate, and scale databases in the cloud. Choose from seven popular engines. Connect to an RDS cluster using IAM authentication."
```
[the remark this answers is not in the document]
Comment (Sizk) [reply lambda-function-connect-rds-iam/snippet-data.json]: Modified

**Review by kathvish — APPROVED** [review]
Thanks for updating!

**Comment by boyney123** [comment]
Thank you for the snippet and @kathvish for reviewing! Will get this merged.

**Pull request merged**